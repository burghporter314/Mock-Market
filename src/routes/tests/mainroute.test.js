import React from 'react';
import MainPage from '../Main';
import renderer from 'react-test-renderer';
import { useAuth0 } from '@auth0/auth0-react';

jest.mock("@auth0/auth0-react");

beforeEach(() => {
    useAuth0.mockReturnValue({
        isAuthenticated: true,
        user: {
            nickname: "testuser"
        },
        logout: jest.fn(),
        loginWithRedirect: jest.fn()
    })
})

it('should match snapshot', () => {

    const component = renderer.create(
            <MainPage/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})