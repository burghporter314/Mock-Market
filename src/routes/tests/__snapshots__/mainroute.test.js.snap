// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  <div
    className="jumbotron"
    style={
      Object {
        "margin": "30px",
      }
    }
  >
    <div
      className="container main-form"
    >
      <form>
        <h2>
          Net Account Value: 
        </h2>
        <hr
          className="colorgraph"
        />
        <div
          className="row"
        >
          <div
            className="form-group col-md-6"
          >
            <label>
              Stock Value
            </label>
            <input
              className="form-control"
              id="inputCity"
              readOnly={true}
              type="text"
            />
          </div>
          <div
            className="form-group col-md-6"
          >
            <label>
              Funds Remaining
            </label>
            <input
              className="form-control"
              id="inputZip"
              readOnly={true}
              type="text"
            />
          </div>
        </div>
        <hr />
        <div
          className="row"
        >
          <div
            className="form-group col-md-4"
          >
            <label>
              Total Profit
            </label>
            <input
              className="form-control"
              id="inputCity"
              readOnly={true}
              type="text"
            />
          </div>
          <div
            className="form-group col-md-4"
          >
            <label>
              Total Deposit
            </label>
            <input
              className="form-control"
              id="inputZip"
              readOnly={true}
              type="text"
            />
          </div>
          <div
            className="form-group col-md-4"
          >
            <label>
              % Change from Deposit
            </label>
            <input
              className="form-control"
              id="inputZip"
              readOnly={true}
              type="text"
            />
          </div>
        </div>
      </form>
    </div>
  </div>,
  <br />,
  <div
    className="jumbotron"
  >
    <div
      className="container table-form"
    >
      <form>
        <h2>
          Transactions 
        </h2>
        <hr
          className="colorgraph"
        />
        <table
          className="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                Stock Ticker
              </th>
              <th>
                Bought price
              </th>
              <th>
                Current Price
              </th>
              <th>
                Amount Bought
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Sample
              </td>
              <td>
                .70
              </td>
              <td>
                .10
              </td>
              <td>
                5
              </td>
            </tr>
            <tr>
              <td>
                Sample2
              </td>
              <td>
                .80
              </td>
              <td>
                .10
              </td>
              <td>
                5
              </td>
            </tr>
          </tbody>
        </table>
      </form>
    </div>
  </div>,
]
`;
